### PR TITLE
Persistent volume support

### PR DESCRIPTION
**Pull request by mudler**
This work is a port of https://github.com/cloudfoundry-incubator/cf-operator/tree/eirini-webhook adapted for eirini-extensions, plus some refactoring and more unit tests added along the way.

It also enables travis to run unit-tests and fixes couple of small issues arised to get to that point.

### CODE REVIEW

File: pkg/controllers/controllers_test.go [hunk 1]
```diff
@@ -19,26 +19,28 @@ import (
...
-			kinds := []string{}
-			for k, _ := range scheme.AllKnownTypes() {
-				kinds = append(kinds, k.Kind)
-			}
-			Expect(kinds).To(ContainElement("BOSHDeployment"))
-			Expect(kinds).To(ContainElement("ExtendedSecret"))
-			Expect(kinds).To(ContainElement("ExtendedStatefulSet"))
-		})
-	})
+	// We don't have any controller right now
```
Comment (mudler) [inline pkg/controllers/controllers_test.go]: Not sure what to do here - we will have them in the future?

File: pkg/kube/webhooks/eirini/volume_webhook.go [hunk 2]
```diff
@@ -0,0 +1,60 @@
+package eirini
```
Comment (viovanov) [inline pkg/kube/webhooks/eirini/volume_webhook.go]: maybe don't include the eirini dir here: https://github.com/SUSE/eirini-extensions/tree/7a64d9f5f0d20643c3475d641ab48e2d0fa40d6c/pkg/kube/webhooks

and have the package name be `webhooks`

then `eirini.Volume(..)` becomes `webhooks.Volume(..)` which makes a bit more sense I think

wdyt?

File: pkg/kube/webhooks/eirini/volume_mutator_test.go [hunk 3]
```diff
@@ -0,0 +1,295 @@
...
+	"code.cloudfoundry.org/cf-operator/pkg/kube/controllers"
+	cfakes "code.cloudfoundry.org/cf-operator/pkg/kube/controllers/fakes"
+	"code.cloudfoundry.org/cf-operator/pkg/kube/util/config"
+	helper "code.cloudfoundry.org/cf-operator/pkg/testhelper"
+	"code.cloudfoundry.org/cf-operator/testing"
+	webhooks "github.com/SUSE/eirini-extensions/pkg/kube/webhooks/eirini"
+)
+
+// labeledPod defines a pod with labels and with VCAP_SERVICES environment set
+func labeledPod(name string, labels map[string]string, vcapServices string) corev1.Pod {
```
[the remark this answers is not in the document]
Comment (mudler) [reply pkg/kube/webhooks/eirini/volume_mutator_test.go]: Oh I see, good point :+1:  will create one for our use cases here ( and try to reuse/upstream the cf-operator catalog )

File: pkg/kube/webhooks/eirini/volume_webhook.go (hunk 2 above)
Comment (mudler) [reply pkg/kube/webhooks/eirini/volume_webhook.go]: :+1:  Good idea. Was actually wondering the same, at the end this project will host only eirini-related code

File: pkg/controllers/controllers_test.go (hunk 1 above)
Comment (mudler) [reply pkg/controllers/controllers_test.go]: Yup, wanted to do it in a separate PR to address all the other areas that needs names refactoring, here I mostly focused to have the test suite running

**Pull request merged**